maint/local_python/binding_c.py

```diff
@@ -1801,7 +1801,7 @@ def dump_validation(func, t):
         if RE.match(r'mpi_(i?m?probe|i?recv|recv_init)$', func_name, re.IGNORECASE) or name == "recvtag":
             # allow MPI_ANY_SOURCE, MPI_PROC_NULL
             G.out.append("MPIR_ERRTEST_RECV_RANK(%s, %s, mpi_errno);" % (comm_ptr, name))
-        elif RE.match(r'(dest|target_rank)$', name) or RE.match(r'mpi_win_', func_name, re.IGNORECASE):
+        elif RE.match(r'(source|dest|target_rank)$', name) or RE.match(r'mpi_win_', func_name, re.IGNORECASE):
```

Comment:
Does this allow `MPI_ANY_SOURCE`?

Reply:
For the functions that allow `MPI_ANY_SOURCE`, it should be captured by the first if-branch. If it leaks to here, then we need fix (the first condition).

I am not sure whether `sendrecv` accepts `MPI_ANY_SOURCE` ...

Reply:
MPI_Sendrecv contains a recv operation. It allows `MPI_ANY_SOURCE`.

Reply:
What I mean is, the receive in `MPI_Sendrecv` is no different than a normal `MPI_Recv`. It just happens concurrently with the send.

Reply:
done.